### DISCUSSION TITLE
Time Series Data Plotting Feature

### DISCUSSION THREAD

**Discussion by mmtaksuu** [alandefreitas/matplotplusplus#171] [Q&A]
Hi guys, I am pretty new on the Matplot++ library and its backend interface. So, I want to ask you about my problem that I encounter while plotting time series data in a loop. I just want to plot time and measurement data as real time by using this library. My problem is that the figure opens and closes very quickly.  How can I solve this problem. I added my code snippet below, if you see any problem in my code please correct me.

Here is my code
```c++
#include <random>
#include <vector>
#include <matplot/matplot.h>

int main()
{
    std::mt19937 eng{std::random_device{}()};
    std::uniform_real_distribution<double> dist{0.0, 500.0};

    std::vector<double> time_vec{};
    std::vector<double> meas_vec{};

    using namespace matplot;

    tiledlayout(1,1);
    auto ax = nexttile();

    ax->xlabel("Time (sn)");
    ax->ylabel("Measurement (m)");
    //ax->xlim({0, 200});
    ax->ylim({0, 500});
    ax->title("Real Time Data Plotting");
    ax->grid(on);
    ax->hold(on);

    for (int i = 0; i < 200; ++i)
    {
        time_vec.push_back(i+1);
        meas_vec.push_back(dist(eng));
        
        ax->clear();
        ax->plot(time_vec, meas_vec, "bo-");
        ax->draw();
    }

    wait();

    return 0;
}
```

Here is my screen record
![time_series](https://user-images.githubusercontent.com/18046031/122987360-30682f00-d3a9-11eb-83ce-08c6d8ce6cfe.gif)

**Comment by alandefreitas** [comment]
Matplot++ will not look good with many frames per second at the moment because the process pipes are very slow so the plot will blink too much. If you don't need that many frames per second, you can update the drawing less often or plot the figures in "quiet" mode, where the figure is only updated when you explicitly call `draw()`. If you really need many frames per second, then it's easier to use a low level graphics library or a library for GUIs.

**Reply by mmtaksuu** [reply]
I tried to update the figure in a seconds. Result was better than now, but its not enough for me. Thanks for your answer. I think I need to use a GUI library like QT.
